internal/log/log.go

```diff
@@ -24,121 +24,115 @@ import (
-) (zap.Logger, error) {
-
-	output := zap.Output(os.Stdout)
-	if c.LogSink != "" {
-		sink, err := os.OpenFile(c.LogSink, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0644)
-		if err != nil {
-			return nil, err
-		}
-		bootstrapLogger.Info("Log file path provided", zap.String("path", c.LogSink))
-		if c.StdOut || foreground {
-			output = zap.Output(zap.MultiWriteSyncer(os.Stdout, sink))
-		} else {
-			output = zap.Output(sink)
-		}
-	}
+func CreateLogger(c *zap.Config, pidPath string, removePIDfunc removePIDfunc) (*Logger, error) {
 
-	fields := zap.Fields(
-		zap.String("service_name", service),
-		zap.String("hostname", hostname),
-		zap.Int("PID", os.Getpid()),
-	)
-
-	logger := PIDRemoverLogger{
-		Logger: zap.New(
-			zap.NewJSONEncoder(),
-			c.Level,
-			fields,
-			output),
+	l, err := c.Build()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create logger")
+	}
+	pl := Logger{
+		Logger:    l,
 		PIDPath:   pidPath,
 		RemovePID: removePIDfunc,
 	}
 
-	return logger, nil
+	return &pl, nil
 }
 
-// ResetLogFile keeps the last 'LogFileSizeKeepKB' KB of the log file if the size of the log file
+// ResetLogFiles keeps the last 'LogFileSizeKeepKB' KB of the log file if the size of the log file
 // has exceeded 'LogFileSizeMaxMB' MB within the last 'PollOrchestratorIntervalSuccess' hours.
-func ResetLogFile(logFilePath string, fileSizeMaxMB int, fileSizeKeepKB int, logger zap.Logger) error {
-	file, err := os.Open(logFilePath)
-	if err != nil {
-		logger.Error("failed to open existing log file", zap.String("file", logFilePath), zap.Error(err))
-		return err
-	}
-	defer file.Close()
+func ResetLogFiles(paths []string, fileSizeMaxMB int, fileSizeKeepKB int, logger *Logger) error {
 
-	// Get the file size
-	stat, err := file.Stat()
-	if err != nil {
-		logger.Error("failed to read the FileInfo structure of the log file",
-			zap.String("file", logFilePath),
-			zap.Error(err))
-		return err
-	}
-	fileSize := int(stat.Size())
-
-	if fileSize > fileSizeMaxMB*1024*1024 {
-		logger.Debug("Size of log file is larger than maximum allowed. Resetting.",
-			zap.String("file", logFilePath),
-			zap.Int("current_size_MB", fileSize),
-			zap.Int("maximum_allowed_size_MB", fileSizeMaxMB))
-
-		buf := make([]byte, fileSizeKeepKB*1024)
-		start := stat.Size() - int64(fileSizeKeepKB*1024)
-		if _, err = file.ReadAt(buf, start); err != nil {
-			logger.Error("failed to read existing log file",
-				zap.String("file", logFilePath),
+	var errs error
+
+	for _, path := range paths {
+		switch path {
+		case "stdout":
+			continue
+		case "stderr":
+			continue
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			logger.Error("failed to open existing log file",
+				zap.String("file", path),
+				zap.Error(err))
+			errs = multierror.Append(errs, err)
+			continue
+		}
+		defer file.Close()
+
+		// Get the file size
+		stat, err := file.Stat()
+		if err != nil {
+			logger.Error("failed to read the FileInfo structure of the log file",
```

Comment:
ResetLogFiles() now does not return anything. It's returned error was not previously used anyways.